### PR TITLE
account abstraction setup

### PR DESCRIPTION
**Pull request by InoMurko**
## Overview

This PR brings in two AA projects: 
https://github.com/eth-infinitism/account-abstraction (contracts)
https://github.com/eth-infinitism/bundler (service)
The purpose of this is making sure things will work in our monorepo as we work on extending `the bundler` and adapting to new functionality on aa (contracts).

## Changes

- add two packages in package/boba
- adjust the dependecies
- fix stale dependencies, fix stale tests, adjust tests to london hf (we were running istanbul)
- follow up PR will build containers and add a docker compose layer that will allow us to write end2end examples of usage
## Testing

Tests are added both for the service and for the contracts.

### CODE REVIEW

**Comment by InoMurko** [comment]
To help review this (252 changes files) - I suggest the focus on reviewing is on the changed files (not newly added). Getting ourselves familiar with bundler and account abstraction later on.

File: packages/contracts/test/contracts/L1/messaging/L1CrossDomainMessenger.spec.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { expect } from '../../../setup'
 /* External Imports */
 import { ethers } from 'hardhat'
 import { Signer, ContractFactory, Contract, BigNumber } from 'ethers'
-import { smockit, MockContract } from '@eth-optimism/smock'
+import { smock, FakeContract } from '@defi-wonderland/smock'
```
Comment (souradeep-das) [inline packages/contracts/test/contracts/L1/messaging/L1CrossDomainMessenger.spec.ts:6]: did the package not work?

File: packages/contracts/test/contracts/L1/messaging/L1StandardBridge.spec.ts [hunk 2]
```diff
@@ -1,81 +1,56 @@
...
 
-const ERR_INVALID_MESSENGER = 'OVM_XCHAIN: messenger contract unauthenticated'
-const ERR_INVALID_X_DOMAIN_MSG_SENDER =
-  'OVM_XCHAIN: wrong sender of cross-domain message'
-const ERR_ALREADY_INITIALIZED = 'Contract has already been initialized.'
-const DUMMY_L2_ERC20_ADDRESS = ethers.utils.getAddress('0x' + 'abba'.repeat(10))
-const DUMMY_L2_BRIDGE_ADDRESS = ethers.utils.getAddress(
-  '0x' + 'acdc'.repeat(10)
-)
+// TODO: Maybe we should consider automatically generating these and exporting them?
```
Comment (souradeep-das) [inline packages/contracts/test/contracts/L1/messaging/L1StandardBridge.spec.ts:10]: generating the error strings?

**Review by souradeep-das — APPROVED** [review]
🚀 🚀

File: packages/contracts/test/contracts/L1/messaging/L1CrossDomainMessenger.spec.ts (hunk 1 above)
Comment (InoMurko) [reply packages/contracts/test/contracts/L1/messaging/L1CrossDomainMessenger.spec.ts:6]: it did work, but it was deprecated by upstream - no more support

Comment (souradeep-das) [reply packages/contracts/test/contracts/L1/messaging/L1CrossDomainMessenger.spec.ts:6]: got it!

**Review by wsdt — APPROVED** [review]
wow! love it

**Pull request merged**